test/CodeGenILTests/ReplicationTests.cs

```diff
@@ -138,5 +137,119 @@ public void MSIL_ReplicationGuides_BuiltinMethods()
+                {
+                    if (!AreEqual(aIEnum[ii], bIEnum[ii]))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+            else if (isAIEnum || isBIEnum)
+            {
+                return false;
+            }
+
+            var aR = a as FFITarget.ReplicationTestA;
+            var bR = a as FFITarget.ReplicationTestA;
+            if (aR != null && bR != null)
+            {
+                return aR.X == bR.X &&
+                       aR.Y == bR.Y &&
+                       aR.Z == bR.Z;
+            }
+            return false;
+        }
+
+        [Test]
+        public void MSIL_FuncCall_Double_SomeGuides()
+        {
+            string code =
+            @"
+            import(""DesignScriptBuiltin.dll"");
+            import (""FFITarget.dll"");
+            x = [0.0,1.0];
+            y = [2.0,3.0];
+            z = [4.0,5.0 ];
+            test1 = FFITarget.ReplicationTestA.ReplicationTestA(x, y, z);  
+            test2 = FFITarget.ReplicationTestA.ReplicationTestA(0,0,0);
+            test3 = FFITarget.ReplicationTestA.ReplicationTestA(x<1>, y<2>, z); 
+            test4 = FFITarget.ReplicationTestA.foo(x, y<2>, z);      
+            ";
+
+            var ast = ParserUtils.Parse(code).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+
+            Assert.IsTrue(AreEqual(
+                new Object[] {
+                    new FFITarget.ReplicationTestA() { X = 0, Y = 2, Z = 4 },
+                    new FFITarget.ReplicationTestA() { X = 1, Y = 3, Z = 5 }
+                },
+                output["test1"]));
+
+            Assert.IsTrue(AreEqual(
+                new Object[] { new FFITarget.ReplicationTestA() { X = 0, Y = 0, Z = 0 } },
+                output["test2"]));
+
+            Assert.IsTrue(AreEqual(
+                new Object[] {
+                    new Object[] {
+                        new Object[] {
+                            new FFITarget.ReplicationTestA() { X = 0, Y = 2, Z = 4 },
+                            new FFITarget.ReplicationTestA() { X = 1, Y = 2, Z = 5 }
+                        },
+                        new Object[] {
+                            new FFITarget.ReplicationTestA() { X = 0, Y = 3, Z = 4 },
+                            new FFITarget.ReplicationTestA() { X = 1, Y = 3, Z = 5 }
+                        }
+                    },
+                    new Object[] {
+                        new Object[] {
+                            new FFITarget.ReplicationTestA() { X = 1, Y = 2, Z = 4 },
+                            new FFITarget.ReplicationTestA() { X = 1, Y = 2, Z = 5 }
+                        },
+                        new Object[] {
+                            new FFITarget.ReplicationTestA() { X = 1, Y = 3, Z = 4 },
+                            new FFITarget.ReplicationTestA() { X = 1, Y = 3, Z = 5 }
```

Comment:
I think the result should be:
0,2,4
0,2,5

0,3,4
0,3,5

1,2,4
1,2,5

1,3,4
1,3,5

Reply:
yes...that looks correct.
I'll take a look

Reply:
the test assertion function was wrong. Fixed directly in branch